Dockerfile

```diff
@@ -523,11 +523,11 @@ ADD patches/sitecustomize.py /root/.local/lib/python3.6/site-packages/sitecustom
 # Set backend for matplotlib
 ENV MPLBACKEND "agg"
 
-# Set up pip to enable pip install.
-ADD patches/kaggle_bashrc /root
-# Patch the system-wide bashrc file for non-root users.
-RUN cat /root/kaggle_bashrc >> /etc/bash.bashrc
-RUN rm /root/kaggle_bashrc
+# Set up an initialization script to be executed before any other commands initiated by the user.
+ADD patches/entrypoint.sh /root/entrypoint.sh
+RUN chmod +x /root/entrypoint.sh
+# This script gets executed by "docker run <image> <command>" and it runs <command> at the end of its execution. 
+ENTRYPOINT ["/root/entrypoint.sh"]
```

Comment:
What did our `ENTRYPOINT` use to be (inherited from our base images)?

Reply:
We may not have one in this Dockerfile, but our image still has one:

```
# docker run -it kaggle/python
root@2d4221a57e4d:/# ps
  PID TTY          TIME CMD
    1 pts/0    00:00:00 tini
    6 pts/0    00:00:00 bash
    7 pts/0    00:00:00 ps
root@2d4221a57e4d:/# exit
root@byod-n1-standard-2-1k80-4br4:/home/kaggle_admin# docker inspect kaggle/python | grep -A3 Entrypoint
            "Entrypoint": null,
            "OnBuild": [],
            "Labels": null
        },
--
            "Entrypoint": [
                "/usr/bin/tini",
                "--"
            ],
```

We very likely want to preserve `tini` as the entrypoint given the functionality it intends to provide.